operatorcourier/flatten.py

```diff
@@ -22,99 +24,101 @@ def get_flattened_files_info(source_dir: str) -> [(str, str)]:
     of operator bundles (CRD, CSV, package) in separate version directories
     :return: A list of tuples where in each tuple, the first element is
     the source file path to be copied to the flattened directory, and the second is
-    the new file name to be used in the file copy. The function returns an empty list if
-    the source_dir is already flat
+    the new file name to be used in the file copy.
     """
-
-    # extract package file and version folders from source_dir
-    root, folder_names, file_names = next(os.walk(source_dir))
-    if not folder_names:
+    root_path, dir_names, root_dir_files = next(os.walk(source_dir))
+    package_content, contains_csv = get_package_csv_info_from_root(source_dir)
+
+    dir_paths = [os.path.join(source_dir, dir_name) for dir_name in dir_names]
+    manifest_paths = list(filter(lambda x: is_manifest_folder(x), dir_paths))
```

Comment:
Can we add a comment here about this code? We had ` extract package file and version folders from source_dir`, and it was removed without any comment explaining what this code does now.